src/journal/journald-server.c

```diff
@@ -2051,10 +2052,12 @@ int server_init(Server *s) {
         if (r < 0)
                 return r;
 
-        /* /dev/ksmg */
-        r = server_open_dev_kmsg(s);
-        if (r < 0)
-                return r;
+        /* /dev/kmsg */
+        if (s->read_kmsg) {
+                r = server_open_dev_kmsg(s);
```

Comment:
`journald` uses `s->dev_kmsg_fd` for writing too. I don't think `ReadKmsg` should affect sending messages to `/dev/kmsg`.